articles/130_ros_time.md

```diff
@@ -70,18 +81,19 @@ For convenience in these cases we will also provide the same API as above, but u
 
 `SystemTime` will be directly tied to the system clock.
 
+#### ROS Time
+
+The `ROSTime` will report the same as `SystemTime` when an ROS Time Source is not active.
+When the ROS time source is active `ROSTime` will return the latest value reported by the Time Source.
+`ROSTime` is considered active when the parameter `using_sim_time` is set on the node.
```

Comment:
Compared to a central parameter in ROS 1 this will be more challenging to achieve when using a node-specific parameter. How would it be configured / ensured that all nodes use sim time?

Reply:
I think likely we'll need a cannonical way to assert these sort of things. Whether it be a node that just stands by and listens for new nodes to come online and sets the parameter on them. As well as some syntax for launch files etc to say apply this parameter to "all" nodes launched.

Reply:
I would say until that is the case I wouldn't "lock in" the way this configuration flag is defined. Maybe just mention that the boolean flag if sim time should be used needs to be retrieved from "somewhere" e.g. a global parameter or a node specific parameter which needs to be kept in sync across multiple nodes.

Reply:
I specifically scoped this to call out the local requirement. As we don't have any guaranteed global connectivity relying on global state before initializing Time is a big problem on startup. Otherwise we'd have to implement some sort of try to attempt to access a global parameter with a non-trivial timeout to support a full discovery cycle. And behavior would change based on potential network topology/connectivity events.

This is both the current implementation and I believe what we should expect for the future.

The API has significantly more interactions for callbacks for changes in time so that if the coordination of setting the parameter is not exactly synchronized as to when it's notified to use sim time it can deal with the changes in a principled manner.

Reply:
I think it's ok to say the local parameter needs to be set. We could additionally initialize it from command line parameters, an environment variable, some global config file, or a global parameter server. In any case, we would read it from the local parameter, most likely.